src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.WorkspaceHost.cs

```diff
@@ -26,11 +33,13 @@ private class WorkspaceHost : ForegroundThreadAffinitizedObject, IVisualStudioWo
 
             public WorkspaceHost(
                 VisualStudioWorkspaceImpl workspace,
-                RemoteHostClient client)
+                RemoteHostClient client,
+                ReferenceCountedDisposable<Connection> currentConnection)
             {
                 _workspace = workspace;
                 _client = client;
                 _currentSolutionId = workspace.CurrentSolution.Id;
+                _currentConnection = new ReferenceCountedDisposable<Connection>.WeakReference(currentConnection);
```

Comment:
❕ Need to handle the case where `currentConnection` is null, either here or in the code that calls this constructor